src/maker/rpc/messages.rs

```diff
@@ -56,6 +50,30 @@ pub enum RpcMsgReq {
     SyncWallet,
 }
 
+/// Represents balance of each utxo type.
+pub struct Balance {
+    /// Seed balance.
+    pub regular: Amount,
+    /// Incoming swap balance.
+    pub swap: Amount,
+    /// Unfinished timelock contract balance.
+    pub contract: Amount,
+    /// Amount locked in Fidelity bonds.
+    pub fidelity: Amount,
+    /// Spendable amount in wallet (seed + swap balance).
+    pub spendable: Amount,
+}
+
+impl Display for Balance {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        write!(
+            f,
+            "Seed balance: {}, Swap balance: {}, Contract balance: {}, Fidelity balance: {}, Spendable balance: {}",
+            self.regular.to_sat(), self.swap.to_sat(), self.contract.to_sat(), self.fidelity.to_sat(), self.regular.to_sat() + self.swap.to_sat()
+        )
+    }
+}
```

Comment:
We can have pretty printout in json like format instead of one line. checkout bitcoin core `getbalances` rpc output. it will be similar.

Also lets not have a display impl. We can format the output at maker-0cli main function. Also the display needs to be different from takers and makers. For takers there is no fidelity balance.

Reply:
yes the first one. You can remove the `balances` key, as that's obvious that these are balances.